tidy3d/plugins/dispersion/fit.py

```
@@ -448,9 +489,144 @@ def plot(
+
+        if has_k > 1:
+            raise ValidationError("Invalid URL. Too many k labels.")
+
+    @classmethod
+    def from_url(cls, url_file: str, delimiter=","):
+        """loads ``DispersionFitter`` from url pointing to a csv/txt file
+            that contains wavelength (micron), n, and optionally k data.
+            Preferred from refractiveindex.info
+
+            The data should be in this format:
+                wl     n
+                float  float
+                .      .
+                .      .
+                .      .
+            (if lossy)
+                wl     k
+                float  float
+                .      .
+                .      .
+                .      .
+
+            The delimiter can be either ',' (csv file), or '\t' (txt file)
+
+        Parameters
+        ----------
+        url_file : str
+            Url link to the csv file.
+            e.g. "https://refractiveindex.info/data_csv.php?datafile=data/main/Ag/Johnson.yml"
+        delimiter : str, optional
+            For refractiveindex.info, delimiter="," for csv, and "\t" for txt
+        """
+
+        # obj to return when failed
+        obj_failed = cls(wvl_um=np.array([1.0]), n_data=np.array([1.0]))
+
+        resp = requests.get(url_file)
+        if resp != 200:  # failed fetching
+            try:
+                resp.raise_for_status()
+            except Exception as e:  # pylint:disable=broad-except
+                log.warning(e)
+                log.error("Connection to the website failed. Please provide a valid URL.")
+                return obj_failed
+
+        data_url = list(
+            csv.reader(codecs.iterdecode(resp.iter_lines(), "utf-8"), delimiter=delimiter)
+        )
+        data_url = list(data_url)
+
+        # first validate data
+        try:
+            cls.validate_url_load(data_url)
+        except ValidationError as e:  # pylint:disable=unused-variable
+            # log.error(e)
+            return obj_failed
```

what is this case?  maybe we should just raise a Tidy3D exception from e?